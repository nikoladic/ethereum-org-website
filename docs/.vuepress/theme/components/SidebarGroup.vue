<template>
  <ul
    ref="items"
    class="sidebar-group-items ma-0 pa-0 no-list"
    v-if="open || !collapsable"
  >
    <li v-for="child in item.children">
      <SidebarLink :item="child" />
    </li>
  </ul>
</template>

<script>
import SidebarLink from './SidebarLink.vue'
import DropdownTransition from './DropdownTransition.vue'

export default {
  name: 'SidebarGroup',
  props: ['item', 'first', 'open', 'collapsable'],
  components: { SidebarLink, DropdownTransition }
}
</script>
