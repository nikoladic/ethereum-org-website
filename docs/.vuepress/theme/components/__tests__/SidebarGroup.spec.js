import { shallowMount } from '@vue/test-utils'
import SidebarGroup from "@/theme/components/SidebarGroup"

describe('SidebarGroup', () => {
  test('render item with children, group is open', () => {
    const wrapper = shallowMount(SidebarGroup, {
      propsData: {
        item: {
          title: "group",
          children: ["a", "b"]
        },
        open: true,
        collapse: false
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render item without children, group is open', () => {
    const wrapper = shallowMount(SidebarGroup, {
      propsData: {
        item: {
          title: "group",
          children: []
        },
        open: false,
        collapse: true
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

