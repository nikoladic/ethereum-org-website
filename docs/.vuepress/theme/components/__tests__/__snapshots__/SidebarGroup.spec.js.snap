// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarGroup render item with children, group is open 1`] = `
<div
  class="sidebar-group"
>
  <p
    class="sidebar-heading open"
  >
    <span>
      group
    </span>
     
    <!---->
  </p>
   
  <dropdowntransition-stub>
    <ul
      class="sidebar-group-items"
    >
      <li>
        <sidebarlink-stub
          item="a"
        />
      </li>
      <li>
        <sidebarlink-stub
          item="b"
        />
      </li>
    </ul>
  </dropdowntransition-stub>
</div>
`;

exports[`SidebarGroup render item without children, group is closed 1`] = `
<div
  class="sidebar-group"
  collapse="true"
>
  <p
    class="sidebar-heading"
  >
    <span>
      group
    </span>
     
    <!---->
  </p>
   
  <dropdowntransition-stub>
    <ul
      class="sidebar-group-items"
    />
  </dropdowntransition-stub>
</div>
`;
